packages/app/src/app/app.component.ts

```diff
@@ -60,12 +60,12 @@ export class AppComponent {
   }
 
   /**
-   * Check if the main app content should be rendered
+   * Check if the main app content should be rendered - we now use only render
+   * if the user is authorized to prevent unstable behaviour with queries
+   * failing
    */
   shouldRenderMainContent(): boolean {
```

Comment:
This works, but normally I would try to use router system and route guards.
https://angular.dev/guide/routing/route-guards

Though I can't decide if this is arguably better. In some ways this is cleaner than URL based state management, e.g. redirecting to `/login?redirect?/location-selection/1` if not authenticated. then on login it fallows the `redirect` param

Or instead of redirect, could return a Promise that resolves when user logs in?
https://angular.dev/api/router/CanActivateFn

But the current design works, so I can't think of a good reason to try to make more use of the Router system, not worth the effort if it doesn't add value

Reply:
Okay - thanks Arlo - good to keep in mind if auth routing needs become more significant/complex